### PR TITLE
Support InternVL3.5 model

### PR DESCRIPTION
**Pull request by fjaijf**
### PR 类型

新增模型支持 / Models

### PR 变更

新增 InternVL3.5-1B 在 PaddleFormers 中的模型接入，包括配置、模型、Processor、ImageProcessor、Auto 注册、单测与本地真实权重验证。

### 主要内容

1. 新增 `paddleformers/transformers/internvl3_5`
   - `InternVLChatConfig`
   - `InternVisionConfig`
   - `InternVLChatModel`
   - `InternVisionModel`
   - `InternVLProcessor`
   - `InternVLImageProcessor`

2. 接入 Auto 入口
   - `AutoConfig`
   - `AutoModel`
   - `AutoProcessor`
   - `AutoImageProcessor`

3. 支持从原始 HuggingFace 权重加载
   - 本地验证模型路径：`/sda/housaijie/code/InternVL3_5-1B`
   - 支持 `convert_from_hf=True`
   - 已验证 `model.safetensors` 权重可完整加载
   - PaddleFormers 加载日志显示所有 checkpoint 权重均被使用

4. 复用主线 Qwen3 / Qwen3-MoE 实现
   - dense 结构复用 `Qwen3ForCausalLMDeprecated`
   - MoE 结构复用 `Qwen3MoeForCausalLMDeprecated`
   - 未重新实现 Qwen3 decoder / attention
   - 单测覆盖 Qwen3-MoE config dispatch 与 router gate 权重转换

5. 新增测试
   - `tests/transformers/internvl3_5/test_modeling.py`
   - `tests/transformers/internvl3_5/test_processor.py`

#### 1. 单卡前向精度对齐

使用本地真实 InternVL3.5-1B 权重，对齐 transformers/PyTorch 与 PaddleFormers 的 FP32 forward logits。

固定输入：

- 模型路径：`/sda/housaijie/code/InternVL3_5-1B`
- 图像输入：`pixel_values shape = (1, 3, 448, 448)`
- 文本输入：`Describe the image.`
- 图像 token 数：`256`
- `<IMG_CONTEXT>` token id：`151671`

结果：

```text
logits shape: (1, 262, 151936)
logits mean abs diff: 5.262756531010382e-05
logits p95 abs diff: 0.00017404556274414062
logits p99 abs diff: 0.00036716461181640625
logits max abs diff: 0.002592325210571289

满足 1e-2 量级要求。

2. 生成结果对齐
使用 greedy search，对比前 10 个生成 token。

torch tokens:
[1096, 2168, 7952, 311, 387, 264, 49400, 476, 49400, 2319]

paddle tokens:
[1096, 2168, 7952, 311, 387, 264, 49400, 476, 49400, 2319]

first10_equal: True

3. 单测结果
执行命令：
> [!NOTE]
>  miniconda3/envs/paddleformers/bin/python -m unittest \
  tests.transformers.internvl3_5.test_modeling \
  tests.transformers.internvl3_5.test_processor

### CODE REVIEW

File: paddleformers/transformers/internvl3_5/configuration.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        if vision_config is None:
+            vision_config = {"architectures": ["InternVisionModel"]}
+        if llm_config is None:
+            llm_config = {"architectures": ["Qwen3ForCausalLM"]}
+
+        self.vision_config = InternVisionConfig(**vision_config) if isinstance(vision_config, dict) else vision_config
+        self.llm_config = Qwen3Config(**llm_config) if isinstance(llm_config, dict) else llm_config
```
Comment (risemeup1111) [inline paddleformers/transformers/internvl3_5/configuration.py]: P1 这里把所有 `llm_config` 都构造成 dense `Qwen3Config`，但公开的 `OpenGVLab/InternVL3_5-30B-A3B` 和 `OpenGVLab/InternVL3_5-241B-A28B` 配置声明的是 `model_type: qwen3_moe` / `Qwen3MoeForCausalLM`。随后 `InternVLChatModel.__init__` 又固定实例化 `Qwen3ForCausalLMDeprecated`，因此这些官方 InternVL3.5 检查点会创建 dense 层，expert/router 权重无法匹配。请按 `llm_config.model_type` 或 `architectures` 分派 Qwen3/Qwen3Moe 的配置、模型及转换规则；若本次只支持 dense 变体，也应在这里显式拒绝不支持的架构，避免静默构造错误模型。

File: paddleformers/transformers/internvl3_5/modeling.py [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+        pos_embed = pos_embed.astype("float32").reshape(
+            [1, self.image_size // self.patch_size, self.image_size // self.patch_size, -1]
+        )
+        pos_embed = pos_embed.transpose([0, 3, 1, 2])
+        pos_embed = F.interpolate(pos_embed, size=[height, width], mode="bicubic", align_corners=False)
+        return pos_embed.reshape([1, -1, height * width]).transpose([0, 2, 1]).astype(target_dtype)
+
+    def forward(self, pixel_values):
+        target_dtype = self.patch_embedding.weight.dtype
+        patch_embeds = self.patch_embedding(pixel_values)
```
Comment (risemeup1111) [inline paddleformers/transformers/internvl3_5/modeling.py:94]: P1 `InternVLImageProcessor` 固定返回 `float32` 的 `pixel_values`，而官方 InternVL3.5 配置带有 `torch_dtype: bfloat16`，`from_pretrained` 会据此以 bf16 创建这里的卷积权重。当前输入未经转换就进入 `Conv2D`，所以 `model.generate(**processor(...))` 在真实 bf16 检查点上会遇到输入与权重 dtype 不一致；仓库内其他 VLM 也都在视觉 patch embedding 前按权重 dtype 转换。请先将 `pixel_values` 转成 `target_dtype`，并增加一个 bf16 模型接收处理器输出的集成测试。

File: paddleformers/transformers/internvl3_5/processor.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        if text is None:
+            data = dict(image_inputs)
+            return BatchFeature(data=data)
+        if not isinstance(text, list):
+            text = [text]
+        text = self._expand_image_tokens(text.copy(), num_patches_list)
+
+        text_inputs = self.tokenizer(text, **output_kwargs["text_kwargs"], return_tensors=return_tensors)
+        data = dict(text_inputs)
+        data.update(dict(image_inputs))
```
Comment (risemeup1111) [inline paddleformers/transformers/internvl3_5/processor.py:108]: P1 `image_inputs` 中包含 `num_patches_list`，这里会把它原样放进处理器的最终 `BatchFeature`；但 `InternVLChatModel.forward` 不接受该参数，因此标准的 `model(**processor(text=..., images=..., return_tensors="pd"))` 路径会直接报 `unexpected keyword argument 'num_patches_list'`。图像 token 展开完成后该元数据已不再参与前向，请在返回前移除它，或让模型显式接收，并补一个处理器输出直连模型的测试。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
| 序号 | 位置 | 优先级 | 状态 |
| --- | --- | --- | --- |
| 1 | [modeling.py:270](https://github.com/PaddlePaddle/PaddleFormers/pull/4867#discussion_r3755865624) | P1 | ✅ |
| 2 | [modeling.py:92](https://github.com/PaddlePaddle/PaddleFormers/pull/4867#discussion_r3755867339) | P1 | ✅ |
| 3 | [processor.py:97](https://github.com/PaddlePaddle/PaddleFormers/pull/4867#discussion_r3755869850) | P1 | ✅ |
| 4 | [internvl3_5/modeling.py:23](https://github.com/PaddlePaddle/PaddleFormers/pull/4867#discussion_r3759134466) | P1 | ❌ |
| 5 | [internvl3_5/modeling.py:507](https://github.com/PaddlePaddle/PaddleFormers/pull/4867#discussion_r3759136123) | P2 | ✅ |

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

File: paddleformers/transformers/internvl3_5/modeling.py (hunk 2 above)
Comment (risemeup1111) [reply paddleformers/transformers/internvl3_5/modeling.py:94]: 已确认当前提交在视觉 patch embedding 前将 `pixel_values` 转为卷积权重 dtype，并补充了 bf16 回归测试；该问题已解决。

File: paddleformers/transformers/internvl3_5/processor.py (hunk 3 above)
Comment (risemeup1111) [reply paddleformers/transformers/internvl3_5/processor.py:108]: 已确认当前提交在展开图像 token 后移除 `num_patches_list`，最终处理器输出与模型前向参数一致；该问题已解决。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]

File: paddleformers/transformers/internvl3_5/configuration.py (hunk 1 above)
Comment (risemeup1111) [reply paddleformers/transformers/internvl3_5/configuration.py]: 仍未完全解决：`modeling.py:258-272` 的 `transpose_weight_keys` 没有包含 MoE router 的 `gate`。`load_checkpoint_format=""` 从 HF 权重加载时会保留 `[num_experts, hidden_size]`，无法匹配 Paddle Linear 的 `[hidden_size, num_experts]`；请补上 `gate`，并增加该加载路径的 MoE checkpoint 转换测试。

Comment (risemeup1111) [reply paddleformers/transformers/internvl3_5/configuration.py]: 已确认当前提交将 `gate` 加入 `transpose_weight_keys`，并新增 `load_checkpoint_format=""` 从 HF safetensors 回读 router 权重的回归测试；该问题已解决。

**Review by risemeup1111 — APPROVED** [review]

**Review by risemeup1111 — APPROVED** [review]

File: paddleformers/transformers/internvl3_5/modeling.py [hunk 4]
```diff
@@ -0,0 +1,639 @@
...
+from ...nn.lm_head import LMHead as GeneralLMHead
+from ..activations import ACT2FN
+from ..conversion_utils import fuse_param_func
+from ..model_outputs import (
+    BaseModelOutput,
+    BaseModelOutputWithPooling,
+    CausalLMOutputWithPast,
+)
+from ..model_utils import PretrainedModel
+from ..qwen3.modeling import Qwen3ForCausalLMDeprecated
```
Comment (risemeup1111) [inline paddleformers/transformers/internvl3_5/modeling.py:23]: P1 这里的顶层导入会让标准安装无法导入 InternVL3.5：`qwen3.modeling` 无条件导入 `gpt_provider`，而后者在未安装可选 extra `paddlefleet` 时会在模块导入阶段抛出 `ImportError`。`setup.py` 只在 `paddleformers[paddlefleet]` 中声明该依赖，因此即使这里只构造不依赖 Fleet 的 `Qwen3ForCausalLMDeprecated`，`from paddleformers.transformers import InternVLChatModel` 也会失败。请恢复 lazy/guarded import，或把 deprecated 实现从 Fleet provider 的导入链中隔离，并补充无 PaddleFleet 的 import smoke test。

File: paddleformers/transformers/internvl3_5/modeling.py [hunk 5]
```diff
@@ -0,0 +1,639 @@
...
+        input_ids: Optional[paddle.Tensor] = None,
+        attention_mask: Optional[paddle.Tensor] = None,
+        position_ids: Optional[paddle.Tensor] = None,
+        image_flags: Optional[paddle.Tensor] = None,
+        visual_features: Optional[paddle.Tensor] = None,
+        past_key_values=None,
+        labels: Optional[paddle.Tensor] = None,
+        use_cache: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
```
Comment (risemeup1111) [inline paddleformers/transformers/internvl3_5/modeling.py]: P2 这个参数现在被静默忽略：下方 `language_model_kwargs` 未包含 `output_hidden_states`，返回值只是读取底层 `outputs.hidden_states`；当前 `Qwen3ForCausalLMDeprecated` 也不处理该标志，所以显式传入 `True` 仍会得到 `None`。与此同时测试已改成直接构造 `InternVisionModel`，不再覆盖 chat wrapper。请恢复 wrapper 的语言模型 hidden states 行为（或移除该公共参数），并用 `InternVLChatModel` 补充回归测试。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]

File: paddleformers/transformers/internvl3_5/modeling.py (hunk 5 above)
Comment (risemeup1111) [reply paddleformers/transformers/internvl3_5/modeling.py]: 已确认当前提交移除了 `InternVLChatModel` 不支持的 `output_hidden_states` 参数，并将回归测试明确改为 `InternVisionModel`；该问题已解决。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
